packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js

```diff
@@ -63,7 +63,33 @@ class TaintTrackingPlugin extends SourceIastPlugin {
 
     this.addSub(
       { channelName: 'datadog:express:query:finish', tag: HTTP_REQUEST_PARAMETER },
-      ({ query }) => this._taintTrackingHandler(HTTP_REQUEST_PARAMETER, query)
+      ({ query }) => {
+        const iastContext = getIastContext(storage('legacy').getStore())
+        if (!iastContext || !query) return
+
+        if (!iastContext.queryCache) {
+          iastContext.queryCache = new Map()
+        }
+
+        for (const key in query) {
+          const currentValue = query[key]
+          if (!currentValue) continue
+
+          const cachedValue = iastContext.queryCache.get(key)
+
+          if (cachedValue === currentValue) {
+            // Reuse cached tainted value
+            query[key] = cachedValue
+          } else {
+            // Taint new value and cache it
+            const taintedValue = typeof currentValue === 'string'
```

Comment:
if `currentValue` could be an object, the `if` of L80 will be `false` because the object instances are going to be diferent.